### PR TITLE
[4.0] RavenDB-11480 : RQL load with index stored fields

### PR DESCRIPTION
**Pull request by aviv86**

### CODE REVIEW

File: src/Raven.Server/Documents/Queries/FieldsToFetch.cs [hunk 1]
```diff
@@ -135,7 +135,15 @@ ref ignored
                 }
             }
 
-            var extract = indexDefinition.MapFields.TryGetValue(selectFieldName, out var value) && value.Storage == FieldStorage.Yes;
+            var key = selectFieldName.Value.Length == 0 &&
```
Comment (arekpalinski) [inline src/Raven.Server/Documents/Queries/FieldsToFetch.cs]: Can we _try_ to put this condition into a method with some meaningful name? Something that would describe what happened that all those conditions are satisfied.

File: src/Raven.Server/Documents/Queries/Results/QueryResultRetrieverBase.cs [hunk 2]
```diff
@@ -399,6 +399,13 @@ private bool TryGetValue(FieldsToFetch.FieldToFetch fieldToFetch, Document docum
                     }
                 }
 
+                else if (fieldToFetch.CanExtractFromIndex)
+                {
+                    var field = luceneDoc.GetField(fieldToFetch.QueryField.SourceAlias);
```
Comment (arekpalinski) [inline src/Raven.Server/Documents/Queries/Results/QueryResultRetrieverBase.cs:404]: Maybe check it `field == null` and throw exception then (unless we are guaranteed that the field exists at this point)

Comment (aviv86) [reply src/Raven.Server/Documents/Queries/Results/QueryResultRetrieverBase.cs:404]: yes we know we that the field exists at this point

File: src/Raven.Server/Documents/Queries/FieldsToFetch.cs (hunk 1 above)
Comment (aviv86) [reply src/Raven.Server/Documents/Queries/FieldsToFetch.cs]: done

File: src/Raven.Server/Documents/Queries/Results/QueryResultRetrieverBase.cs (hunk 2 above)
Comment (ayende) [reply src/Raven.Server/Documents/Queries/Results/QueryResultRetrieverBase.cs:404]: How do you know that the field must be here?

Comment (arekpalinski) [reply src/Raven.Server/Documents/Queries/Results/QueryResultRetrieverBase.cs:404]: @aviv86 ping

Comment (aviv86) [reply src/Raven.Server/Documents/Queries/Results/QueryResultRetrieverBase.cs:404]: Actually the field can be missing if the index has dynamic fields. fixing that

**Pull request merged**